core/sql/optimizer/SearchKey.cpp

```diff
@@ -2711,6 +2716,65 @@ void SearchKey::computeCoveredLeadingKeys()
   }
 }
 
+void SearchKey::computeFullKeyPredicates(ValueIdSet& predicates)
+{
+  // If the key column is an INDEX COLUMN, convert it into a base column
+  ValueIdList baseKeyCols;
+
+  ValueId vid;
+  const ValueIdList& keyColumns = getKeyColumns();
+  for (CollIndex i=0; i<keyColumns.entries(); i++ ) {
+     if ( keyColumns[i].getItemExpr()->getOperatorType() == ITM_INDEXCOLUMN )
+       vid = ((IndexColumn*)(keyColumns[i].getItemExpr()))->getDefinition();
+     else
+       vid = keyColumns[i];
+
+     baseKeyCols.insertAt(baseKeyCols.entries(), vid);
+  }
```

Comment:
So this code works for index key columns and base table key columns (because if it is a base table key column, we go to line 2730 and don't attempt any mapping).